www/js/matrix.js

```diff
@@ -0,0 +1,166 @@
+        if (i > 0) {
+          preEvent = weekEvents[i-1];
+        } else {
+          preEvent = 'undefined';
+        }
+        setRegisteredStatus(event);
+        setEventProgress(event, preEvent);
+
+        // Modify time if single or double dot
+        event.start = new Date(event.start);
+        event.time = event.start.hhmm();
+        if (event.dot === 'single') {
+          event.time += ' (.)';
+        } else if (event.dot === 'double') {
+          event.time += ' (..)';
+        }
+
+        // Add date and switch side of timeline if new day
+        if (lastDay !== event.start.getDate()) {
+          if (side === 'right') {
+            side = 'left';
+          } else {
+            side = 'right';
+          }
+          matrixEventHTML = app.templates.matrixEventTemplate({date: event.start, side: side, event: event});
+        } else {
+          matrixEventHTML = app.templates.matrixEventTemplate({side: side, event: event});
+        }
+        $(matrixEventHTML).hide().appendTo(timelineList).fadeIn(fadeInTime);
+        lastDay = event.start.getDate();
+      }
+      week += 1;
+    });
+  }
+  nbrIntroductionWeeks = week;
+}
+
+function setEventProgress(event, preEvent) {
+  // Returns if event has already occured, if it is the next
+  // event or if it is further in the future
+  date = new Date();
+  today = date.yyyymmdd();
+  now = today + ' ' + date.hhmm();
+
+  // Day and time of current and previous event
+  eventDate = new Date(event.start);
+  eventDay = eventDate.yyyymmdd();
+  eventTime = eventDate.yyyymmdd() + ' ' + eventDate.hhmm();
+  if (preEvent !== 'undefined') {
+    preEventTime = new Date(preEvent.start);
+    preEventTime = preEventTime.yyyymmdd() + ' ' + preEventTime.hhmm();
+  } else {
+    preEventTime = 'undefined';
+  }
+
+  // Check if event is occuring today
+  if (eventDay === today) {
+    progress = 'today-';
+  } else {
+    progress = '';
+  }
+
+  // Check if event is in the past, up next or further into the future
+  if (eventTime < now) {
+    progress += 'past';
+  } else if ((preEventTime !== 'undefined' && preEventTime < now) || (preEventTime === 'undefined' && eventTime > now)) {
+    progress += 'next';
+  } else {
+    progress += 'future';
+  }
+  event.progress = progress;
+}
+
+function scrollToToday() {
+  // Find today's first event
+  date = $$('.timeline-item-divider.today-past');
+  if (date.length === 0) {
+    date = $$('.timeline-item-divider.today-next');
+    if (date.length === 0) {
+      date = $$('.timeline-item-divider.today-future');
+      if (date.length === 0) {
+        // No event today, look for next event
+        date = $$('.timeline-item-divider.next');
+        if (date.length === 0) {
+          // Current date is after introduction
+          return;
+        }
+      }
+    }
+  }
+  today = date[0];
+  offset = $$('.navbar').height(); // Compensate for navbar
+  pos = today.style.position;
+  today.style.position = 'relative';
```

Comment:
I think the position is relative from the framework, so this could probably be removed.